### PR TITLE
feat(generative-ai): Update Gemini Flask and Pro model names

### PR DESCRIPTION
**Pull request by msampathkumar**
## Description

Update the Gemini Model name's suffix, from 001 to 002 (latest version)

Fixes b/369345741

### CODE REVIEW

**Review by rsamborski — APPROVED** [review]

File: generative_ai/batch_predict/gemini_batch_predict.py [hunk 1]
```diff
@@ -45,9 +47,9 @@ def batch_predict_gemini_createjob(input_uri: str, output_uri: str) -> BatchPred
 
     # Submit a batch prediction job with Gemini model
     batch_prediction_job = BatchPredictionJob.submit(
-        source_model="gemini-1.5-flash-001",
+        source_model="gemini-1.5-flash-002",
```
Comment (holtskinner) [inline generative_ai/batch_predict/gemini_batch_predict.py:50]: Have we confirmed that 002 works with batch prediction?

**Review by holtskinner — APPROVED** [review]

File: generative_ai/batch_predict/gemini_batch_predict.py (hunk 1 above)
Comment (msampathkumar) [reply generative_ai/batch_predict/gemini_batch_predict.py:50]: Short answer. No.

I am testing it for all feature, right now. If required, I am on standby to roll back ASAP.

Comment (gericdong) [reply generative_ai/batch_predict/gemini_batch_predict.py:50]: It's confirmed that batch prediction works with 002 models.

**Comment by msampathkumar** [comment]
New error -  `google.api_core.exceptions.ResourceExhausted: 429 Quota exceeded for aiplatform.googleapis.com/generate_content_requests_per_minute_per_project_per_base_model with base model: gemini-1.5-pro. Please submit a quota increase request. https://cloud.google.com/vertex-ai/docs/generative-ai/quotas-genai.`

**Comment by msampathkumar** [comment]
An error is reported at https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12625#issuecomment-2372439139 & Bug fix is WIP @ https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12630

**Review by gericdong — APPROVED** [review]

**Pull request merged**